### PR TITLE
526 refactor task card component to improve readability and maintainability

### LINKED ISSUE #526: Refactor Task Card Component to Improve Readability and Maintainability
**Description**
The[ src/components/tasks/card/index.tsx ](https://github.com/Real-Dev-Squad/website-status/blob/30225429984dcdcf746fdef59f92705ee4e15f82/src/components/tasks/card/index.tsx)file is becoming too complex with a lot of code stuffed into it. To improve the readability and maintainability of the code, let's refactor the Task Card component and break it down into smaller, more manageable pieces.

Some potential steps we could take to achieve this include:

- Identify logical sections of the component that can be split into separate sub-components.
- Extract repetitive or complex logic into separate helper functions or utilities.
- Simplify the component's interface by removing unnecessary props or state variables.
- Ensure that the component follows best practices for naming, formatting, and commenting.

By breaking down the Task Card component in this way, we can make it easier for developers to work with the code, and reduce the risk of bugs and errors creeping in over time.

### PR DESCRIPTION
**Pull request by TanishqSingla**
### Description:
This PR is intended closes #526 which is a refactor issue

### Anthing you would like to inform the reviewer about:
The PR is following stacked PR pattern, so following are the constituents of the PR
- #610 
- #621
- #627

### Tasks
- [x] Fix folder structure
- [x] Extract component
- [x] Fix logical errors
- [x] Improve readibility of code
- [ ] Fix types

### Dev Tested:
- [ ] Yes

### CODE REVIEW

**Review by khushi818 — DISMISSED** [review]
LGTM

**Review by kotesh-arya — DISMISSED** [review]
Great work on setting up a dedicated folder for the Loader and consistency of Letter-Casing for the Card.

File: src/components/tasks/Card/TaskTagEdit.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React, { useState } from 'react';
 
-import classNames from '@/components/tasks/card/card.module.scss';
+import classNames from '@/components/tasks/Card/card.module.scss';
```
Comment (shubhamsigdar1) [inline src/components/tasks/Card/TaskTagEdit.tsx:3]: card is folder name...it is not component??

**Review by shubhamsigdar1 — COMMENTED** [review]
Great work @TanishqSingla ...please can you reply one of my comment

**Comment by sahsisunny** [comment]
@TanishqSingla Can you add the test coverage screenshot if written?

File: src/components/Loaders/Loader.module.scss
Comment (sahsisunny) [inline src/components/Loaders/Loader.module.scss:1]: Can you change the PX to REM?

**Review by sahsisunny — COMMENTED** [review]
Please add the test coverage and change PX to REM.

**Pull request closed without merging**